pkg/cidata/cidata.TEMPLATE.d/boot/00-check-rtc-and-wait-ntp.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+set -eu
+
+# In vz, the VM lacks an RTC when booting with a kernel image (see: https://developer.apple.com/forums/thread/760344).
+# This causes incorrect system time until NTP synchronizes it, leading to TLS errors.
+# To avoid TLS errors, this script waits for NTP synchronization if RTC is unavailable.
+test ! -c /dev/rtc0 || exit 0
+
+# This script is intended for services running with systemd.
+command -v systemctl >/dev/null 2>&1 || exit 0
+
+# Enable `systemd-time-wait-sync.service` to wait for NTP synchronization at an earlier stage.
+systemctl enable systemd-time-wait-sync.service
+
+# For the first boot, where the above setting is not yet active, wait for NTP synchronization here.
+until ntp_synchronized=$(timedatectl show --property=NTPSynchronized --value) && [ "${ntp_synchronized}" = "yes" ]; do
```

Comment:
Can we trigger synchronous NTP update instead of waiting for unpredictable time?

Reply:
By the time this script is reached, the NTP sync for `systemd-timesyncd.service` has already been triggered. Additionally, `timedatectl` has no option to wait for NTP synchronization, which results in this type of loop.

Reply:
So why it takes sometimes 30 seconds?

Do we have a way to start the instance using the current host time instead of depending on NTP? Since we rebuild the cidata.iso on each reboot, we can inject there the current host time, and have the guest clock starting at the cidata.iso build time. This should be in the worst case few seconds after the real start time.

Reply:
Or maybe better, can we inject the start time using the kernel command line? This is relevant only when using kernel image which we use to allow using kernel command line.